eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1366,6 +1366,11 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, remo
       // they can publish a closing tx with any sig we sent them, even if we are not done negotiating
       handleMutualClose(getMutualClosePublished(tx, d.closingTxProposed), Left(d))
 
+    case Event(WatchFundingSpentTriggered(tx), d: DATA_NEGOTIATING) if d.bestUnpublishedClosingTx_opt.map(_.tx).contains(tx) =>
```

Comment:
I think we should match on `txid`, not on the whole tx, it protects us against witness mutations.
Since this is spending a multisig utxo, they could chose to reorder the two signatures and your check on tx equality would fail.

Reply:
Good point, and also it wasn't consistent with the other handlers. Fixed with 0d4b436.